### PR TITLE
Separate betweenness and closeness; new compute algorithms

### PR DESCRIPTION
**Pull request by kylegranger**
This should be the last big change to betweenness and closeness computations: swap out the old inner loop compute code, and swap in new & much improved algorithms.  This builds on the correctness and multi-threading of the previous PR.
- handle betweenness and closeness computations separately, no more tuples returned
   - use the Brandes algorithm for betweenness computation.  It is a standard, and gives us 30-80x improvement in efficiency
   - for closeness, use a straight forward BFS algorithm, this takes about 20% the time of betweenness.
- new betweenness and closeness modules
- compute module is gone
- the previously #ignore'd test which took 82 seconds, now takes 1.16 seconds with 8 threads, so it is no longer ignored.
- the results from medium (2472 nodes) and large (6729 nodes) samples have been compared to the results computed by another third party library (teexgraph) and match, so are therefore still correct. 
- two smaller tests also match three independent libraries:  teexgraph (C++), a Python library, and a JavaScript library.

### CODE REVIEW

File: src/graph.rs [hunk 1]
```diff
@@ -1164,22 +1156,58 @@ mod tests {
...
+        let mut closeness = [0.0; N];
+        for i in 0..N {
+            betweenness[i] = *between_map.get(&i).unwrap();
+            closeness[i] = *close_map.get(&i).unwrap();
         }
 
+        let total_path_length = [9, 8, 8, 7, 7, 9];
+        let mut expected_closeness = [0.0; N];
+        let mut expected_betweenness = [1.0, 1.5, 1.5, 2.5, 2.5, 0.0];
+        const DIVISOR: f64 = ((N - 1) * (N - 2) / 2) as f64;
```
Comment (asmie) [inline src/graph.rs:1206]: It looks like divisor in 126 line in betweenness.rs file but here is additional division by 2 - line 126 does not have such division - why?

File: src/graph.rs [hunk 2]
```diff
@@ -1195,20 +1223,55 @@ mod tests {
...
         let total_path_length = [15, 9, 10, 12, 15, 12, 15];
-        let mut expected_closeness: [f64; 7] = [0.0; 7];
-        let expected_betweenness: [f64; 7] = [0.5, 9.5, 9.0, 2.0, 0.0, 2.0, 0.0];
-        for i in 0..7 {
-            expected_closeness[i] = total_path_length[i] as f64 / 6.0;
+        let mut expected_closeness = [0.0; N];
+        let mut expected_betweenness = [0.5, 9.5, 9.0, 2.0, 0.0, 2.0, 0.0];
+        for i in 0..N {
+            expected_closeness[i] = total_path_length[i] as f64 / (N - 1) as f64;
+            expected_betweenness[i] /= ((N - 1) * (N - 2) / 2) as f64;
```
Comment (asmie) [inline src/graph.rs:1274]: The same question like for DIVISOR const a little above 😃

**Review by asmie — APPROVED** [review]
Looks more clean than previous version.
Separation: 👍 
If the algorithm agrees with three other libraries so I assume that implementation is right - I looked through the code but not verified by hand that algo is producing right results 😸

File: src/graph.rs (hunk 1 above)
Comment (kylegranger) [reply src/graph.rs:1206]: Good to question that!  However, the code it correct:
1. The short answer is, the `/2` is already in the data (`expected_betweenness`).  For example, we have our `randomish_graph` with 7 nodes.  So in the `betweenness.rs` code, we divide the results by 2, or by 30 (6x5).  In the test data though, the `expected_betweenness` contains the non-normalized results, i.e., **already divided-by-two**.  So, we only need to divide that by 15, not 30.  I.e.:  `(N-1)(N-2)/2`
2. I confirmed that the results match what the python library is sending when the normalized flag is set
Screenshot 2023-03-27 at 09 28 02

This is what out test code outputs:
```
compute_closeness: done 62.958µs
  normalized: i 0 = 0.03333333333333333
  normalized: i 1 = 0.6333333333333333
  normalized: i 2 = 0.6
  normalized: i 3 = 0.13333333333333333
  normalized: i 4 = 0
  normalized: i 5 = 0.13333333333333333
  normalized: i 6 = 0
ok
```
I hope this is clear.

Comment (kylegranger) [reply src/graph.rs:1206]: I just checked in a change, so the second test case with normalization uses the DIVISOR same as in the first case.  😄

File: src/graph.rs (hunk 2 above)
Comment (kylegranger) [reply src/graph.rs]: I just checked in a change, so this second test case with normalization uses the DIVISOR same as in the first case. 😄

File: src/graph.rs (hunk 1 above)
Comment (asmie) [reply src/graph.rs:1206]: Yes, it's clear :)

**Review by niklaslong — APPROVED** [review]

**Pull request merged**